api/tests/gmsa_test_client.cpp

```diff
@@ -581,7 +581,19 @@ int parse_credspec_domainless_test(std::string credspec)
     int response = parse_cred_spec_domainless(credspec, krb_ticket_info, krb_ticket_arn_mapping );
     std::cout << krb_ticket_arn_mapping->credential_spec_arn;
     std::cout << krb_ticket_arn_mapping->krb_file_path;
-    return response;
+    if(response == 0)
+    {
+       return true;
+    }
+    return  false;
+}
+
+int validate_domain()
+{
+    return (isValidDomain("a.com") && isValidDomain("ab.toto-abc.com") &&
+             !isValidDomain("p/") && isValidDomain("test4.gmsa-pentest.com") &&
+             !isValidDomain ("-testdomain.org") &&  isValidDomain("contoso.com") &&
+             !isValidDomain(".org"));
```

Comment:
nit: I'd recommend clubbing all valid and invalid cases separately for readability 
```
string validDomains[] = {
    "a.com",
    "ab.toto-abc.com", 
    "test4.gmsa-pentest.com",
    "contoso.com"
  };

string invalidDomains[] = {  
    "p/",
    "-testdomain.org",
    ".org"
  };
``` 
Assert `validDomains` needs to be valid and `invalidDomains` needs to be invalid when `isValidDomain` is called

Reply:
we need to redo all the unit tests with a framework we can take an AI for then